### DISCUSSION TITLE
logging failed attempt in a logger

### DISCUSSION THREAD

**Discussion by BigBoulard** [go-resty/resty#622] [Q&A]
Hi,

I'm searching a way to log the failed attempt when `Resty` tries to reach a service that is not available.
I've found `OnError` but it's triggered once all attempts have been reached.

Thx

**Comment by BigBoulard** [comment]
Just saw that I can do that inside of `AddRetryCondition`. Don't think it was intended to be used this way but it works and I don't have any better solution.

```go
r.AddRetryCondition(
		func(r *resty.Response, err error) bool {
                          if r.StatusCode() > 299 || r.StatusCode() == 0 {
			                          log.L.Warn( // zerolog
				                          err, "main", "httpclient",
				                          fmt.Sprintf("%s %s - %s", r.Request.Method, r.Request.URL, r.Status()),
			                          )
                          }
			return r.Header().Get("Should-Retry") == "true" ||
				(r.StatusCode() == http.StatusTooManyRequests ||
					r.StatusCode() == http.StatusServiceUnavailable ||
					r.StatusCode() == http.StatusGatewayTimeout)
		},
)
```
I noticed that `r.StatusCode()` return `0` and not `503`. I'm pretty ignorant about this, but if 429 is for `too many requests`, 0 is if in case there's no response, then what `503` would be used for then?

```go
func (r *Response) StatusCode() int {
	if r.RawResponse == nil {
		return 0
	}
	return r.RawResponse.StatusCode
}
```

**Comment by jeevatkm** [comment]
Details about 503 - https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503
